snapshotter/cmd/devmapper/main.go

```diff
@@ -30,10 +34,55 @@ const (
 )
 
 func main() {
-	var configPath string
+	var (
+		ctx        = context.Background()
+		config     = &devmapper.Config{}
+		configPath = ""
+		rootPath   = ""
+	)
+
 	flag.StringVar(&configPath, "config", "", "Path to devmapper configuration file")
+	flag.StringVar(&rootPath, "path", "", "Path to snapshotter data")
+
+	// These flags are needed for compatibility with container-storage-setup tool as there is no way to
+	// ignore unknown flags in `flag` package. Storage options are parsed using visitKeyValueOpts func.
+	flag.String("storage-driver", "devicemapper", "Storage driver to use. Always devicemapper.")
+	flag.Bool("storage-opt", false, "Storage configuration options (compatible with Docker dm.* flags)")
+
 	flag.Parse()
 
+	// Try load file from disk
+	if cfg, err := loadConfig(ctx, configPath); err == nil {
+		config = cfg
+	} else if err != os.ErrNotExist {
+		log.G(ctx).WithError(err).Fatal("failed to load config file")
+		return
+	}
+
+	if rootPath != "" {
+		config.RootPath = rootPath
+	}
+
+	// Append and/or overwrite file configuration with --storage-opt dm.XXX=YYY command line flags
+	if err := visitKeyValueOpts("--storage-opt", func(key, value string) error {
+		return applyStorageOpt(ctx, key, value, config)
+	}); err != nil {
+		log.G(ctx).WithError(err).Fatal("failed to apply storage options")
+		return
+	}
+
+	if err := config.Validate(); err != nil {
+		log.G(ctx).WithError(err).Fatal("invalid configuration")
+		return
```

Comment:
same here

Reply:
How do you feel about getting rid of the `return` statements since they don't really do anything.

Reply:
They don't, but I've added them to explicitly state that program is not going to continue execution from this block, because `log.G(ctx) ... ` something might not be clear enough

Reply:
I would much prefer a comment to signify things. I think adding code to signify something when it does nothing will lead to confusion. I was confused at least ><

Reply:
I don't want to write a comment instead of every return, there are log error messages for that.
Also I don't understand how `return` may confuse someone :)